### PR TITLE
MPT-21801 Distribute Saving Plans recurring fee per linked account by covered usage share

### PR DESCRIPTION
**Pull request by ruben-sebrango**
🤖 AI-generated PR — Please review carefully.

## Summary

- Add `SavingsPlanJournalLineProcessor` that skips `SAVING_PLAN_RECURRING_FEE` when split billing is enabled, so the org-level fee is not consumed by the default MPA-routing path.
- Add `SavingPlansDistributionProcessor` that collects the org-level recurring fee and distributes it proportionally to linked accounts weighted by `SavingsPlanCoveredUsage / total_covered` (one journal line per account, rounded to 6 dp).
- Register the new line processor in `journal_line.py` and invoke the distribution processor in `agreement.py` when `split_billing_enabled`.
- Under `MASTER_PAYER` the existing behaviour is preserved: the recurring fee is routed to the MPA subscription unchanged.

## Test plan

- [ ] `make check-all` passes (ruff, flake8, all 1067+ tests, 99% coverage)
- [ ] `test_distributes_sp_fee_proportionally` — proportional amounts for ACC-A (75%) and ACC-B (25%) are correct
- [ ] `test_routes_to_linked_account_subscription` — journal line targets the linked-account subscription, not MPA
- [ ] `test_uses_usage_sku` — USAGE SKU is applied
- [ ] `test_skips_account_with_zero_covered_usage` — accounts with no covered usage are excluded
- [ ] `test_returns_empty_when_no_recurring_fee` — no-op when fee is absent
- [ ] `test_returns_empty_when_no_covered_usage` — no-op when covered usage is absent
- [ ] `test_sums_multiple_recurring_fee_metrics` — multiple recurring fee entries are summed before distributing
- [ ] `test_skips_account_when_distributed_amount_rounds_to_zero` — accounts whose share rounds to zero are excluded
- [ ] `test_skips_recurring_fee_when_split_billing_enabled` — line processor returns empty list under split billing
- [ ] `test_processes_recurring_fee_when_split_billing_disabled` — line processor routes to MPA under MASTER_PAYER
- [ ] `test_routes_to_master_payer_subscription_when_not_split` — subscription search targets MPA vendor ID

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Closes [MPT-21801](https://softwareone.atlassian.net/browse/MPT-21801)

**Release Notes:**

- Implemented distributed billing for AWS Saving Plans recurring fees across linked accounts under the `LINKED_ACCOUNT_PERCENTAGE` split billing policy
- Added `SavingsPlanJournalLineProcessor` to suppress recurring fee processing when split billing is enabled, preventing duplicate billing to the org-level fee
- Added `SavingPlansDistributionProcessor` to collect org-level recurring fees and distribute them proportionally to linked accounts based on their `SavingsPlanCoveredUsage` share
- Generated journal lines allocated per account with amounts rounded to 6 decimal places (one line per account)
- Preserved existing behavior under `MASTER_PAYER` mode where recurring fees continue to route to the MPA subscription through the standard processor
- Registered the new processors in the billing pipeline for seamless integration with existing workflows
- Added comprehensive test coverage for proportional distribution, subscription targeting, SKU application, edge cases, and mode-specific routing behavior

[MPT-21801]: https://softwareone.atlassian.net/browse/MPT-21801?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: swo_aws_extension/billing/generators/additional_line_processors/saving_plans.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        for account_id, account_usage in usage_result.usage_by_account.items():
+            amount = sum(
+                metric.amount
+                for metric in account_usage.get_metrics_by_record_type(
+                    AWSRecordTypeEnum.SAVING_PLAN_COVERED_USAGE
+                )
+            )
+            if amount > DEC_ZERO:
+                covered[account_id] = amount
+        return covered
```
[the remark this answers is not in the document]
Comment (ruben-sebrango) [reply swo_aws_extension/billing/generators/additional_line_processors/saving_plans.py:116]: If the MPA has a savings plan usage, the usage should be included in the MPA subscription

**Review by alephsur — APPROVED** [review]
